### PR TITLE
Misc Fixes from Router Refactor

### PR DESCRIPTION
**Pull request by koonsies**
## What's this PR do?

Cleans up some promise calls and removes unused code.

## How should this be manually tested?

Check to see that the apps in question still work. 
(1) Reporting
(2) Manage Users
(3) Nonuser Outreach

## Any background context you want to provide?
Sometimes I write code bad

### CODE REVIEW

File: gulp/src/reporting/components/SetUpReport.jsx [hunk 1]
```diff
@@ -468,7 +472,7 @@ class SetUpReport extends Component {
       });
     }
 
-    return (
+    return pageLoading ? <LoadingSpinner/> : (
```
Comment (aljordan82) [inline gulp/src/reporting/components/SetUpReport.jsx]: The loadingSpinner overlay doesn't cover all page.

Comment (aljordan82) [inline gulp/src/reporting/components/SetUpReport.jsx]: Never mind, I'll fix this in another branch.

File: gulp/src/manageusers/components/Role.jsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Role extends React.Component {
       return iRole.role_id === parseInt(roleId, 10);
     })[0];
     const assignedActivities = {};
-    Object.keys(props.activities).forEach((key) => {return assignedActivities[key];});
+    Object.keys(props.activities).map(key => assignedActivities[key] = []);
```
Comment (koonsies) [inline gulp/src/manageusers/components/Role.jsx]: This was a change during the lint-fix PR that broke the Role view on Manage Users

File: gulp/src/manageusers/components/Role.jsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ class Role extends React.Component {
   componentDidMount() {
     const {dispatch} = this.props;
     dispatch(clearValidationAction());
-    (async () => this.initialLoad())(); // add then/catch handling here
+    this.initialLoad(); // add then/catch handling here
```
Comment (koonsies) [inline gulp/src/manageusers/components/Role.jsx]: I was wrapping this in fancy handling without benefit. Cleaning up code.

File: gulp/src/nonuseroutreach/components/ProcessRecordPage.jsx [hunk 4]
```diff
@@ -63,9 +63,8 @@ class ProcessRecordPage extends Component {
     const {dispatch, match} = this.props;
     dispatch(setPageAction('process'));
     dispatch(markPageLoadingAction(true));
-    (async() => {
-      await dispatch(doLoadEmail(match.params.recordId));
-    })().then(() => dispatch(markPageLoadingAction(false)));
+    dispatch(doLoadEmail(match.params.recordId))
```
Comment (koonsies) [inline gulp/src/nonuseroutreach/components/ProcessRecordPage.jsx]: Same as above

File: gulp/src/reporting/components/DynamicReportApp.jsx [hunk 5]
```diff
@@ -1,56 +1,16 @@
...
 import ReportingIntention from './ReportingIntention';
-import {
-  doLoadReportSetUp,
-  doDataSetMenuFill,
-} from '../actions/compound-actions';
 
 
 class DynamicReportApp extends Component {
-  componentDidMount() {
-    const {dispatch} = this.props;
```
Comment (koonsies) [inline gulp/src/reporting/components/DynamicReportApp.jsx]: Most unused code here

File: gulp/src/reporting/components/DynamicReportApp.jsx [hunk 6]
```diff
@@ -98,13 +58,8 @@ class DynamicReportApp extends Component {
...
-  history: PropTypes.object.isRequired,
-  dispatch: PropTypes.func.isRequired,
   reportFinder: PropTypes.object.isRequired,
   location: PropTypes.object,
 };
 
-export default connect(s => ({
-  pageLoading: s.loading.mainPage,
-}))(DynamicReportApp);
+export default DynamicReportApp;
```
Comment (koonsies) [inline gulp/src/reporting/components/DynamicReportApp.jsx]: Didn't need the store in this component

File: gulp/src/reporting/components/ExportReport.jsx [hunk 7]
```diff
@@ -35,8 +35,7 @@ class ExportReport extends Component {
     } else {
       dispatch(highlightReportAction());
     }
-    (async () => this.loadData())()
-      .then(dispatch(markPageLoadingAction(false)));
+    this.loadData().then(() => dispatch(markPageLoadingAction(false)));
```
Comment (koonsies) [inline gulp/src/reporting/components/ExportReport.jsx]: Removed unneeded fancy-wrap

File: gulp/src/reporting/components/SetUpReport.jsx [hunk 8]
```diff
@@ -87,10 +88,12 @@ class SetUpReport extends Component {
     } = queryObject;
     const reportDataId = Number.parseInt(reportDataIdString, 10);
     const {currentFilter, name} = location.state || {};
+
+    // both must be complete before loading spinner is removed
     (async () => {
       await dispatch(doDataSetMenuFill(intention, category, dataSet));
       await dispatch(doLoadReportSetUp(reportDataId, currentFilter, name));
-    })().then(dispatch(markPageLoadingAction(false)));
+    })().then(() => dispatch(markPageLoadingAction(false)));
```
Comment (koonsies) [inline gulp/src/reporting/components/SetUpReport.jsx]: Fix fancy wrap to give a callable function, rather than a called function

**Pull request merged**